site/src/pages/WorkspacePage/ResourcesSidebar.tsx

```diff
@@ -0,0 +1,109 @@
+import { Interpolation, Theme } from "@emotion/react";
+import Skeleton from "@mui/material/Skeleton";
+import { useTheme } from "@mui/material/styles";
+import { WorkspaceResource } from "api/typesGenerated";
+import {
+  Sidebar,
+  SidebarCaption,
+  SidebarItem,
+} from "components/FullPageLayout/Sidebar";
+import { getResourceIconPath } from "utils/workspace";
+
+type ResourcesSidebarProps = {
+  failed: boolean;
+  resources: WorkspaceResource[];
+  onChange: (resourceId: string) => void;
+  selected: string;
+};
+
+export const ResourcesSidebar = (props: ResourcesSidebarProps) => {
+  const theme = useTheme();
+  const { failed, onChange, selected, resources } = props;
+
+  return (
+    <Sidebar>
+      <SidebarCaption>Resources</SidebarCaption>
+      {failed && (
+        <p
+          css={{
+            margin: 0,
+            padding: "0 16px",
+            fontSize: 13,
+            color: theme.palette.text.secondary,
+            lineHeight: "1.5",
+          }}
+        >
+          Your workspace build failed, so the necessary resources couldn&apos;t
+          be created.
+        </p>
+      )}
+      {resources.length === 0 &&
+        !failed &&
```

Comment:
Is this condition strictly for handling loading states?

Reply:
Yes.